pyccel/ast/f2py.py

```diff
@@ -121,7 +124,7 @@ def as_static_function(func, name=None):
     int_kind = Variable('int', 'f2py_array_dimension').precision
 
     for a in args:
-        if a.rank > 1 and a.order == 'C':
+        if rank and a.rank > 1 and a.order == 'C':
```

Comment:
Is `rank` correctly defined here?

Reply:
it's not correctly defined. do you think it's good to have `FunctionAddress` inherits from `Variable` so we can avoid this kind of checks, in this case for a `FunctionAddress` the rank would be 0, it's as if we are treating it as a c pointer.

Reply:
I think we already discussed this and decided that it was not quite a `Variable`. That said given the number of places in the code where you need to do `ifinstance(a, (Variable, FunctionAddress))` I am not sure I agree. What do you think @rmahjoubi @yguclu ? As a minimum even if it doesn't inherit from `Variable` I think it should inherit from `PyccelAstNode`

Reply:
I think so

Reply:
My comment here originally was not about whether `rank` is correctly defined with regards to `FunctionAddress` but whether the variable `rank` in this line has meaning.
The previous time that you use rank you define it (https://github.com/pyccel/pyccel/pull/502/files#diff-0243435837f0679be53be6bdf9816fa4b1f5d7cd3410affd18f0e7508a9726f0R77-R80) however we are now in a new loop and `rank` still has the value of the last argument of the previous loop.
This could be why your tests are failing

Reply:
in the recent commit, i got rid of the variable `rank` completely, now i'am using `a.rank` only where a is the variable.

Reply:
:+1: yes I saw, I posted my comment a few seconds before your commit ^^'
We can resolve this conversation once the issue about FunctionAddress and Variable has been created

Reply:
I just created it